### PR TITLE
Makefile: build/test app on mac

### PR DESCRIPTION
**Pull request by ajinkyaraj-23**
sed command removed as it does not support `\U` on Mac (bsd rather than gnu version).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -41,27 +41,28 @@ docker_images:	docker_speculos			\
 		docker_ledger_app_integration_tests
 
 scan-build-%:
-	SDK=$(shell echo $@ | sed 's/scan-build-\(.*\)/\U\1/')_SDK;	\
+	SDK=$(shell echo $* | tr "[:lower:]" "[:upper:]")_SDK;   \
```
Comment (emturner) [inline Makefile:44]: nit: backlashes no longer aligned

File: Makefile [hunk 2]
```diff
@@ -41,27 +41,28 @@ docker_images:	docker_speculos			\
...
 
-app_%.tgz:	app/src/*.[ch]		\
+app_%_dbg.tgz:	app/src/*.[ch]		\
 		app/src/parser/*.[ch]	\
 		app/Makefile
-	SDK=$(shell echo $@ | sed 's/app_\(.*\).tgz/\U\1/')_SDK;   \
-	$(DOCKER_RUN_APP_BUILDER) bash -c                          \
-            "BOLOS_SDK=\$$$$SDK make -C app"
+	SDK=$(shell echo $* | tr "[:lower:]" "[:upper:]")_SDK;   \
+	$(DOCKER_RUN_APP_BUILDER) bash -c                            \
```
Comment (emturner) [inline Makefile:55]: nit: backslashes no longer aligned

File: Makefile (hunk 1 above)
Comment (emturner) [reply Makefile:44]: (multiple places in the Makefile). Otherwise LGTM

File: Makefile [hunk 3]
```diff
@@ -35,45 +35,46 @@ docker_ledger_app_integration_tests:
...
 	$(DOCKER_RUN_APP_BUILDER) make -C app mrproper
 	$(DOCKER_RUN_APP_OCAML) bash -c "make -C /app/tests/generate clean && cd /app && rm -rf _build"
 
-unit_tests:	test/samples/micheline/nano/samples.hex	\
-		test/samples/operations/nano/samples.hex	\
-		tests/unit/*.ml*			\
-		tests/unit/*.[ch]			\
-		tests/unit/dune				\
+unit_tests:	test/samples/micheline/nanos/samples.hex	\
+		test/samples/operations/nanos/samples.hex	\
```
Comment (spalmer25) [inline Makefile:74]:
```suggestion
unit_tests:	test/samples/micheline/nano/samples.hex	\
		test/samples/operations/nano/samples.hex	\
```

**Comment by ajinkyaraj-23** [comment]
After making required changes, 
I am getting one test failure in integration tests.
Please find the test result attached. 
[integration_tests.json](https://github.com/trilitech/ledger-app-tezos-wallet/files/13176661/integration_tests.json)

**Comment by spalmer25** [comment]
> After making required changes, I am getting one test failure in integration tests. Please find the test result attached. [integration_tests.json](https://github.com/trilitech/ledger-app-tezos-wallet/files/13176661/integration_tests.json)

From the json, it seems that the `$COMMIT_BYTES` variable has not been initialised correctly.
And indeed, a variable has been moved in PR https://github.com/trilitech/ledger-app-tezos-wallet/pull/93 in which a `sed` has been moved in `app_vars.sh`.

File: Makefile [hunk 4]
```diff
@@ -35,33 +35,34 @@ docker_ledger_app_integration_tests:
...
-	SDK=$(shell echo $@ | sed 's/app_\(.*\).tgz/\U\1/')_SDK;   \
-	$(DOCKER_RUN_APP_BUILDER) bash -c                          \
-            "BOLOS_SDK=\$$$$SDK make -C app"
+	scan-build-nanox scan-build-stax
+
+app_%_dbg.tgz:	app/src/*.[ch]	\
+	app/src/parser/*.[ch]	\
+	app/Makefile
+	SDK=$(shell echo $* | tr "[:lower:]" "[:upper:]")_SDK;	\
+	$(DOCKER_RUN_APP_BUILDER) bash -c						\
```
Comment (emturner) [inline Makefile:55]:
```suggestion
	$(DOCKER_RUN_APP_BUILDER) bash -c			\
```

File: Makefile [hunk 5]
```diff
@@ -35,33 +35,34 @@ docker_ledger_app_integration_tests:
...
-app_%_dbg.tgz:	app/src/*.[ch]		\
+
+app_%.tgz:	app/src/*.[ch]		\
 		app/src/parser/*.[ch]	\
 		app/Makefile
-	SDK=$(shell echo $@ | sed 's/app_\(.*\)_dbg.tgz/\U\1/')_SDK; \
-	$(DOCKER_RUN_APP_BUILDER) bash -c                            \
-            "BOLOS_SDK=\$$$$SDK make -C app DEBUG=1"
+	SDK=$(shell echo $* | tr "[:lower:]" "[:upper:]")_SDK;	\
+	$(DOCKER_RUN_APP_BUILDER) bash -c						\
```
Comment (emturner) [inline Makefile:64]:
```suggestion
	$(DOCKER_RUN_APP_BUILDER) bash -c			\
```

**Comment by emturner** [comment]
v v minor whitespace - can be fixed in future :)

**Review by emturner — CHANGES_REQUESTED** [review]
The gitconfig change breaks the tests on linux

**Review by emturner — APPROVED** [review]

**Pull request merged**